src/event-emitter.js

```diff
@@ -12,6 +12,12 @@ define(['lodash-amd/modern/arrays/pull'], function (pull) {
   EventEmitter.prototype.on = function (eventName, fn) {
     var listeners = this._listeners[eventName] || [];
 
+    listeners.forEach(function(listener){
+      if(listener === fn){
+        return;
+      }
+    });
```

Comment:
I don't think this does what you expect. You're just returning from the function you passed to `forEach`, not the parent function `forEach` is called from. [`Array.prototype.some`](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/some) may be more useful here.